onnx/defs/tensor/defs.cc

```diff
@@ -170,6 +171,23 @@ Example 2:
             "Constrain input and output types to float tensors.");
 
 ONNX_OPERATOR_SCHEMA(Transpose)
+    .TensorInferenceFunction([](Node* n) {
```

Comment:
Hmm such in-place interface looks to (maybe just) me another styling issue, e.g. you can easily forget to check `out->sizes().empty()`. What would be better is a more functional interface that takes the operator and inputs' shapes and return output's shapes.

Reply:
yeah, I can probably do that. It does require exposing a little more of ir.h to schema.h though

Reply:
yeah, we should have more functional interface I'd say. Also, there are probably some easy helper functions we can do to help writing shape inference functions

Reply:
yes, all standard abstraction/code reuse strategies apply, since they're just functions